From: Michael Buesch <mb@bu3sch.de>
Date: Fri, 22 Jun 2007 22:13:00 +0000 (+0200)
Subject: bcm43xx-mac80211: Fix build for PCI-less systems.
X-Git-Url: http://bu3sch.de/gitweb?p=wireless-dev.git;a=commitdiff_plain;h=edd2b9cc18cf2eb15ea35d34af556b614f07114b

bcm43xx-mac80211: Fix build for PCI-less systems.

pci_iomap() and pci_iounmap() are not available if PCI
support is disabled.

Signed-off-by: Michael Buesch <mb@bu3sch.de>
---

diff --git a/drivers/ssb/scan.c b/drivers/ssb/scan.c
index feaf1e5..b5d909c 100644
--- a/drivers/ssb/scan.c
+++ b/drivers/ssb/scan.c
@@ -202,7 +202,11 @@ void ssb_iounmap(struct ssb_bus *bus)
 		iounmap(bus->mmio);
 		break;
 	case SSB_BUSTYPE_PCI:
+#ifdef CONFIG_SSB_PCIHOST
 		pci_iounmap(bus->host_pci, bus->mmio);
+#else
+		assert(0); /* Can't reach this code. */
+#endif
 		break;
 	}
 	bus->mmio = NULL;
@@ -222,7 +226,11 @@ static void __iomem * ssb_ioremap(struct ssb_bus *bus,
 		mmio = ioremap(baseaddr, SSB_CORE_SIZE);
 		break;
 	case SSB_BUSTYPE_PCI:
+#ifdef CONFIG_SSB_PCIHOST
 		mmio = pci_iomap(bus->host_pci, 0, ~0UL);
+#else
+		assert(0); /* Can't reach this code. */
+#endif
 		break;
 	}
 
